src/mixins/subscribe-mixin.ts

```diff
@@ -0,0 +1,49 @@
+import { LitElement, Constructor, PropertyValues } from "lit-element";
+import { UnsubscribeFunc } from "home-assistant-js-websocket";
+
+export interface HassSubscribeElement {
+  hassSubscribe(): UnsubscribeFunc[];
+}
+
+/* tslint:disable-next-line */
+export const SubscribeMixin = <T extends LitElement>(
+  superClass: Constructor<T>
+): Constructor<T & HassSubscribeElement> =>
+  // @ts-ignore
+  class extends superClass {
+    private _unsubs?: UnsubscribeFunc[];
+
+    public connectedCallback() {
+      super.connectedCallback();
+      this._subscribe();
+    }
+
+    public disconnectedCallback() {
+      super.disconnectedCallback();
+      if (this._unsubs) {
+        while (this._unsubs.length) {
+          this._unsubs.pop()!();
+        }
+        this._unsubs = undefined;
+      }
+    }
+
+    protected updated(changedProps: PropertyValues) {
+      super.updated(changedProps);
+      if (changedProps.has("hass")) {
```

Comment:
We should add a `static get properties()` that returns `{ hass: {} }`. Otherwise `hass` might not trigger a re-render, never ending up calling our `updated` function.